pkg/controller/upgradeconfig/cluster_upgrader.go

```diff
@@ -418,23 +424,24 @@ func RemoveMaintWindow(c client.Client, m maintenance.Maintenance, upgradeConfig
 // This perform cluster health check after upgrade
 func PostClusterHealthCheck(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
 
-	return performClusterHealthCheck(c)
+	return performClusterHealthCheck(c, upgradeConfig)
 }
 
 // Check whether nodes are upgraded or not
-func NodesUpgraded(c client.Client, nodeType string, reqLogger logr.Logger) (bool, error) {
+func NodesUpgraded(c client.Client, nodeType string, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
```

Comment:
Do we need to pass the entire UpgradeConfig object into this func when we only reference a string ? `upgradeConfig.Name`